bundles/org.openhab.binding.zwavejs/src/main/java/org/openhab/binding/zwavejs/internal/handler/ZwaveJSNodeHandler.java

```diff
@@ -0,0 +1,718 @@
+            } else {
+                updateState(Objects.requireNonNull(colorCapability.colorTempChannel), UnDefType.UNDEF);
+            }
+        }
+
+        return newState;
+    }
+
+    @Override
+    public void onNodeDead(Event event) {
+        logger.trace("Node {}. Dead", config.id);
+        updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "@text/offline.comm-error.dead-node");
+    }
+
+    @Override
+    public void onNodeAlive(Event event) {
+        logger.trace("Node {}. Alive", config.id);
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public Integer getId() {
+        return this.config.id;
+    }
+
+    private boolean setupThing(Node node) {
+        logger.debug("Node {}. Building channels and configuration, containing {} values", node.nodeId,
+                node.values.size());
+
+        configurationAsChannels = Objects.requireNonNull(getBridge()).getConfiguration()
+                .as(ZwaveJSBridgeConfiguration.class).configurationChannels;
+
+        ZwaveJSTypeGeneratorResult result;
+        try {
+            result = typeGenerator.generate(thing.getUID(), node, configurationAsChannels);
+        } catch (Exception e) {
+            logger.error("Node {}. Error generating type information", node.nodeId, e);
+            return false;
+        }
+
+        ThingBuilder builder = editThing();
+
+        // Update location if needed
+        if (!result.location.equals(getThing().getLocation()) && !result.location.isBlank()) {
+            builder.withLocation(result.location);
+        }
+
+        // Update channels
+        builder = updateChannels(builder, result);
+
+        colorCapabilities = result.colorCapabilities;
+        if (logger.isDebugEnabled()) {
+            colorCapabilities.forEach((e, c) -> logger.debug("Node {}. Endpoint {}, {}", node.nodeId, e, c));
+        }
+
+        updateThing(builder.build());
+
+        // Initialize state for channels and configuration
+        initializeChannelAndConfigState(node, result);
+
+        return true;
+    }
+
+    private ThingBuilder updateChannels(ThingBuilder builder, ZwaveJSTypeGeneratorResult result) {
+        List<Channel> channelsToRemove = new ArrayList<>();
+        for (Channel channel : thing.getChannels()) {
+            if (!result.channels.containsKey(channel.getUID().getId())) {
+                channelsToRemove.add(channel);
+            } else {
+                result.channels.remove(channel.getUID().getId());
+            }
+        }
+        if (!channelsToRemove.isEmpty()) {
+            logger.trace(null, "Node {}. Removing {} channels", this.config.id, channelsToRemove.size());
+            builder.withoutChannels(channelsToRemove);
+        }
+        if (!result.channels.isEmpty()) {
+            List<Channel> channels = result.channels.entrySet().stream().sorted(Map.Entry.comparingByKey())
+                    .map(Map.Entry::getValue).collect(Collectors.toList());
```

Comment:
You can replace `.collect(Collectors.toList())` by `.toList()`

Reply:
Thanks